public/js/components/Expressions.js

```diff
@@ -3,66 +3,111 @@ const { connect } = require("react-redux");
 const { bindActionCreators } = require("redux");
 const ImPropTypes = require("react-immutable-proptypes");
 // const classnames = require("classnames");
+const Svg = require("./utils/Svg");
 const actions = require("../actions");
 const { getExpressions, getPause } = require("../selectors");
-// const ObjectInspector = React.createFactory(require("./ObjectInspector"));
+const Rep = React.createFactory(require("./Rep"));
 // const { truncateStr } = require("../utils/utils");
 const { DOM: dom, PropTypes } = React;
 
 require("./Expressions.css");
 
 const Expressions = React.createClass({
   propTypes: {
-    pauseInfo: ImPropTypes.map,
     expressions: ImPropTypes.list,
     addExpression: PropTypes.func,
     updateExpression: PropTypes.func,
-    evaluateExpression: PropTypes.func,
-    loadObjectProperties: PropTypes.func,
-    command: PropTypes.func
+    deleteExpression: PropTypes.func
   },
 
   displayName: "Expressions",
 
   addExpression(e) {
+    const { addExpression } = this.props;
     if (e.key === "Enter") {
       const expression = {
         input: e.target.value
       };
       if (e.target.id) {
         expression.id = e.target.id.split("-").pop();
       }
-      this.props.addExpression(expression);
+      e.target.value = "";
+      addExpression(expression);
     }
   },
 
   updateExpression(e) {
+    const { updateExpression } = this.props;
+    const id = e.target.id.split("-").pop();
     const expression = {
-      id: e.target.id.split("-").pop(),
-      input: e.target.textContent.split(" --> ")[0]
+      id,
+      input: e.target.textContent
     };
-    this.props.updateExpression(expression);
+    updateExpression(expression);
+    setTimeout(() => {
```

Comment:
could the `setTimeout` go away if the focus call were in the right lifecycle method `componentDidUpdate` maybe?